packages/utils/src/query/metadata.ts

```diff
@@ -1,11 +1,17 @@
 // Copyright 2020-2022 OnFinality Limited authors & contributors
 // SPDX-License-Identifier: Apache-2.0
 
+import {blake2AsHex} from '@polkadot/util-crypto';
+
 export const METADATA_REGEX = /^_metadata$/;
 
 export const MULTI_METADATA_REGEX = /^_metadata_[a-zA-Z0-9-]+$/;
 
-export function getMetadataTableName(chainId: string): string {
-  // 63 chars is the max postgres database table name length
-  return `_metadata${chainId ? `_${chainId.substring(0, 63)}` : ''}`;
+export function getMetadataTableName(chainId: string | undefined): string {
+  if (chainId) {
+    const hash = blake2AsHex(chainId);
```

Comment:
change to use  64 bytes  `const hash = blake2AsHex(chainId,64)`; the output hash length will be 18.